### PR TITLE
DxDiagram new API

### PR DESCRIPTION
**Pull request by LanaDX**

### CODE REVIEW

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapeToolboxTemplate.md [hunk 1]
```diff
@@ -4,16 +4,25 @@ type: template
...
 ##### field(data.item): dxDiagramShape
-<!-- Description goes here -->
+Information about the currently processed shape.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+The template content must be presented as SVG elements.
+
+Use the [toolboxTemplate](/api-reference/10%20UI%20Widgets/dxDiagram/1%20Configuration/customShapes/toolboxTemplate.md '/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/customShapes/#toolboxTemplate') option to define a template of an individual shape in the toolbox.
```
Comment (albertov05) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapeToolboxTemplate.md]: Can you clarify what you mean by "a template of an individual shape"?

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapes/toolboxWidthToHeightRatio.md [hunk 2]
```diff
@@ -4,7 +4,9 @@ type: Number
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies an aspect ratio of the shape in the toolbox.
```
Comment (albertov05) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapes/toolboxWidthToHeightRatio.md]:
```suggestion
Specifies the aspect ratio of the shape in the toolbox.
```

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapes/toolboxWidthToHeightRatio.md [hunk 3]
```diff
@@ -4,7 +4,9 @@ type: Number
...
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+The size of a shape in the toolbox is determined automatically based on the toolbox width ([width](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/toolbox/#width) and the number of shapes per row ([shapeIconsPerRow](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/toolbox/#shapeIconsPerRow).
```
Comment (albertov05) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapes/toolboxWidthToHeightRatio.md]: Just checking, but I wanted to make sure you're using the correct number of parenthesis in this sentence. Do you need an additional parenthesis after the `width` url and the `shapeIconsPerRow` url?

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapes/toolboxWidthToHeightRatio.md [hunk 4]
```diff
@@ -4,7 +4,9 @@ type: Number
...
 ##### shortDescription
-<!-- Description goes here -->
+Specifies an aspect ratio of the shape in the toolbox.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+The size of a shape in the toolbox is determined automatically based on the toolbox width ([width](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/toolbox/#width) and the number of shapes per row ([shapeIconsPerRow](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/toolbox/#shapeIconsPerRow). 
+
+Use the **toolboxWidthToHeightRatio** option to change an aspect ratio of the shape.
```
Comment (albertov05) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapes/toolboxWidthToHeightRatio.md]:
```suggestion
Use the **toolboxWidthToHeightRatio** option to change the aspect ratio of the shape.
```

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/edges/customDataExpr.md [hunk 5]
```diff
@@ -5,10 +5,14 @@ default: undefined
...
+Specifies the name of a data source field or an expression that provides an edge's custom data.
 
 ##### param(data): Object
-<!-- Description goes here -->
+The current edge's data object.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+This option links custom data contained in a data source to a diagram edge. The linked data is copied and pasted with the edge. Changes in the data are reflected in a diagram history and can be undone and redone by the UI.
```
Comment (albertov05) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/edges/customDataExpr.md]: "The linked data is copied and pasted with the edge."

Can you clarify what you mean here?

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/nodes/customDataExpr.md [hunk 6]
```diff
@@ -5,10 +5,14 @@ default: undefined
...
+Specifies the name of a data source field or an expression that provides a node's custom data.
 
 ##### param(data): Object
-<!-- Description goes here -->
+The current node's data object.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+This option links custom data contained in a data source to a diagram node. The linked data is copied and pasted with the node. Changes in the data are reflected in a diagram history and can be undone and redone by the UI.
```
Comment (albertov05) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/nodes/customDataExpr.md]: "The linked data is copied and pasted with the node."

Can you clarify what you mean here?

File: api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md [hunk 7]
```diff
@@ -3,13 +3,20 @@ id: dxDiagram.reloadContent(itemKey, applyLayout)
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Reloads diagram data from a data source.
 
 ##### param(itemKey?): any
-<!-- Description goes here -->
+Specifies a data key of an item to reload.
```
Comment (albertov05) [inline api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md]:
```suggestion
Specifies the data key of the item to reload.
```

File: api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md [hunk 8]
```diff
@@ -3,13 +3,20 @@ id: dxDiagram.reloadContent(itemKey, applyLayout)
...
+* If the parameter is specified the widget reloads the corresponding item and every added and deleted items.
```
Comment (albertov05) [inline api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md]: Can you clarify what you mean by "every added and deleted items"?

File: api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md [hunk 9]
```diff
@@ -3,13 +3,20 @@ id: dxDiagram.reloadContent(itemKey, applyLayout)
...
+* If the parameter is not specified the widget reloads the entire diagram.
```
Comment (albertov05) [inline api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md]:
```suggestion
* If the parameter is not specified, the widget reloads the entire diagram.
```

File: api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md [hunk 10]
```diff
@@ -3,13 +3,20 @@ id: dxDiagram.reloadContent(itemKey, applyLayout)
...
 
 ##### param(applyLayout?): Boolean
-<!-- Description goes here -->
-
+Specifies whether the widget applies the [auto layout](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/nodes/autoLayout/) after content is reloaded.
 ---
-<!-- Description goes here -->
\ No newline at end of file
+The changes applied by this method are reflected in a diagram history and can be undone and redone by the UI.
```
Comment (albertov05) [inline api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md]:
```suggestion
The changes applied by this method are reflected in the diagram history, and can be undone and redone by the UI.
```

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapeToolboxTemplate.md (hunk 1 above)
Comment (LanaDX) [reply api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapeToolboxTemplate.md]: Diagram has two options for a shape template in the toolbox: one specifies a template for all shapes (diagram.customShapeToolboxTemplate). Another specifies a template for a particular shape (customShape.toolboxTemplate), so users can create a template for every shape.

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapes/toolboxWidthToHeightRatio.md (hunk 3 above)
Comment (LanaDX) [reply api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapes/toolboxWidthToHeightRatio.md]: I can rewrite it as follows:
```suggestion
The size of a shape in the toolbox is determined automatically based on the toolbox [width](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/toolbox/#width and the [number of shapes per row](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/toolbox/#shapeIconsPerRow. 
```

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/edges/customDataExpr.md (hunk 5 above)
Comment (LanaDX) [reply api-reference/10 UI Widgets/dxDiagram/1 Configuration/edges/customDataExpr.md]: A diagram item (node or edge) can contain data that are not visible in the diagram, e.g., detailed information about a person.  If required, a user can access this data directly (without this option) from a data source, but in this case the widget does not copy-pasted the custom data with the item. If user specifies this option, the custom data is linked to an item and a copy-pasted item contain all the custom information.

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/nodes/customDataExpr.md (hunk 6 above)
Comment (LanaDX) [reply api-reference/10 UI Widgets/dxDiagram/1 Configuration/nodes/customDataExpr.md]: the same as above

File: api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md (hunk 8 above)
Comment (LanaDX) [reply api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md]: If the parameter is specified, the widget does not search differences in existing items, but it updates the specified item,  inserted items, and deleted items

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapeToolboxTemplate.md (hunk 1 above)
Comment (albertov05) [reply api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapeToolboxTemplate.md]:
```suggestion
Use the [toolboxTemplate](/api-reference/10%20UI%20Widgets/dxDiagram/1%20Configuration/customShapes/toolboxTemplate.md '/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/customShapes/#toolboxTemplate') option to define a template for each shape in the toolbox.
```

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapes/toolboxWidthToHeightRatio.md (hunk 3 above)
Comment (albertov05) [reply api-reference/10 UI Widgets/dxDiagram/1 Configuration/customShapes/toolboxWidthToHeightRatio.md]: Is my suggestion below correct:

```suggestion
The size of a shape in the toolbox is determined automatically based on the toolbox [width](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/toolbox/#width) and the [number of shapes per row](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/toolbox/#shapeIconsPerRow).
```

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/edges/customDataExpr.md (hunk 5 above)
Comment (albertov05) [reply api-reference/10 UI Widgets/dxDiagram/1 Configuration/edges/customDataExpr.md]: Thank you. Could we say:

```suggestion
This option links custom data from a data source to the diagram edge. The edge contains the linked data copied from the data source. Changes in the data are reflected in the diagram history. You can use the UI to undo and redo these changes.
```

File: api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md [hunk 11]
```diff
@@ -3,13 +3,20 @@ id: dxDiagram.reloadContent(itemKey, applyLayout)
...
+* If the parameter is not specified, the widget reloads the entire diagram.
 
 ##### param(applyLayout?): Boolean
-<!-- Description goes here -->
-
+Specifies whether the widget applies the [auto layout](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/nodes/autoLayout/) after content is reloaded.
 ---
-<!-- Description goes here -->
\ No newline at end of file
+The changes applied by this method are reflected in the diagram history, and can be undone and redone by the UI.
```
Comment (albertov05) [inline api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md]:
```suggestion
Changes applied by this method are reflected in the diagram history. You can use the UI to undo and redo these changes.
```

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/nodes/customDataExpr.md (hunk 6 above)
Comment (albertov05) [reply api-reference/10 UI Widgets/dxDiagram/1 Configuration/nodes/customDataExpr.md]:
```suggestion
This option links custom data from a data source to a diagram node. The node contains the linked data copied from the data source. Changes in the data are reflected in the diagram history. You can use the UI to undo and redo these changes.
```

File: api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md (hunk 8 above)
Comment (albertov05) [reply api-reference/10 UI Widgets/dxDiagram/3 Methods/reloadContent(itemKey_applyLayout).md]:
```suggestion
* If the parameter is specified, the widget reloads the corresponding item, inserted items and deleted items.
```

**Review by AnnaKondratova — APPROVED** [review]

**Pull request merged**